care/facility/api/viewsets/patient.py

```diff
@@ -272,9 +273,11 @@ def filter_queryset(self, request, queryset, view):
             elif view.action != "transfer":
                 allowed_facilities = get_accessible_facilities(request.user)
                 q_filters = Q(facility__id__in=allowed_facilities)
+                if view.action == "retrieve":
+                    q_filters |= Q(consultations__facility__id__in=allowed_facilities)
                 q_filters |= Q(last_consultation__assigned_to=request.user)
                 q_filters |= Q(assigned_to=request.user)
-                queryset = queryset.filter(q_filters)
+                queryset = queryset.filter(q_filters).distinct()
```

Comment:
Can you check the query here? usually postgres does a distinct check on all fields which is extremely inefficient

Reply:
Here's the query that's executed:

<details>
<summary>SQL Query</summary>

```sql
SELECT DISTINCT "facility_patientregistration"."id",
                "facility_patientregistration"."external_id",
                "facility_patientregistration"."created_date",
                "facility_patientregistration"."modified_date",
                "facility_patientregistration"."deleted",
                "facility_patientregistration"."source",
                "facility_patientregistration"."facility_id",
                "facility_patientregistration"."nearest_facility_id",
                "facility_patientregistration"."meta_info_id",
                "facility_patientregistration"."name",
                "facility_patientregistration"."age",
                "facility_patientregistration"."gender",
                "facility_patientregistration"."phone_number",
                "facility_patientregistration"."emergency_phone_number",
                "facility_patientregistration"."address",
                "facility_patientregistration"."permanent_address",
                "facility_patientregistration"."pincode",
                "facility_patientregistration"."date_of_birth",
                "facility_patientregistration"."year_of_birth",
                "facility_patientregistration"."nationality",
                "facility_patientregistration"."passport_no",
                "facility_patientregistration"."is_medical_worker",
                "facility_patientregistration"."blood_group",
                "facility_patientregistration"."contact_with_confirmed_carrier",
                "facility_patientregistration"."contact_with_suspected_carrier",
                "facility_patientregistration"."estimated_contact_date",
                "facility_patientregistration"."past_travel",
                "facility_patientregistration"."countries_travelled_old",
                "facility_patientregistration"."countries_travelled",
                "facility_patientregistration"."date_of_return",
                "facility_patientregistration"."allergies",
                "facility_patientregistration"."present_health",
                "facility_patientregistration"."ongoing_medication",
                "facility_patientregistration"."has_SARI",
                "facility_patientregistration"."is_antenatal",
                "facility_patientregistration"."last_menstruation_start_date",
                "facility_patientregistration"."date_of_delivery",
                "facility_patientregistration"."ward_old",
                "facility_patientregistration"."ward_id",
                "facility_patientregistration"."local_body_id",
                "facility_patientregistration"."district_id",
                "facility_patientregistration"."state_id",
                "facility_patientregistration"."is_migrant_worker",
                "facility_patientregistration"."disease_status",
                "facility_patientregistration"."number_of_aged_dependents",
                "facility_patientregistration"."number_of_chronic_diseased_dependents",
                "facility_patientregistration"."last_edited_id",
                "facility_patientregistration"."action",
                "facility_patientregistration"."review_time",
                "facility_patientregistration"."created_by_id",
                "facility_patientregistration"."is_active",
                "facility_patientregistration"."date_of_receipt_of_information",
                "facility_patientregistration"."test_id",
                "facility_patientregistration"."date_of_test",
                "facility_patientregistration"."srf_id",
                "facility_patientregistration"."test_type",
                "facility_patientregistration"."allow_transfer",
                "facility_patientregistration"."last_consultation_id",
                "facility_patientregistration"."will_donate_blood",
                "facility_patientregistration"."fit_for_blood_donation",
                "facility_patientregistration"."village",
                "facility_patientregistration"."designation_of_health_care_worker",
                "facility_patientregistration"."instituion_of_health_care_worker",
                "facility_patientregistration"."transit_details",
                "facility_patientregistration"."frontline_worker",
                "facility_patientregistration"."date_of_result",
                "facility_patientregistration"."number_of_primary_contacts",
                "facility_patientregistration"."number_of_secondary_contacts",
                "facility_patientregistration"."is_vaccinated",
                "facility_patientregistration"."number_of_doses",
                "facility_patientregistration"."vaccine_name",
                "facility_patientregistration"."covin_id",
                "facility_patientregistration"."last_vaccinated_date",
                "facility_patientregistration"."cluster_name",
                "facility_patientregistration"."is_declared_positive",
                "facility_patientregistration"."date_declared_positive",
                "facility_patientregistration"."assigned_to_id",
                "facility_patientregistration"."abha_number_id",
                "facility_facility"."id",
                "facility_facility"."external_id",
                "facility_facility"."created_date",
                "facility_facility"."modified_date",
                "facility_facility"."deleted",
                "facility_facility"."name",
                "facility_facility"."is_active",
                "facility_facility"."verified",
                "facility_facility"."facility_type",
                "facility_facility"."kasp_empanelled",
                "facility_facility"."features",
                "facility_facility"."longitude",
                "facility_facility"."latitude",
                "facility_facility"."pincode",
                "facility_facility"."address",
                "facility_facility"."ward_id",
                "facility_facility"."local_body_id",
                "facility_facility"."district_id",
                "facility_facility"."state_id",
                "facility_facility"."oxygen_capacity",
                "facility_facility"."type_b_cylinders",
                "facility_facility"."type_c_cylinders",
                "facility_facility"."type_d_cylinders",
                "facility_facility"."expected_oxygen_requirement",
                "facility_facility"."expected_type_b_cylinders",
                "facility_facility"."expected_type_c_cylinders",
                "facility_facility"."expected_type_d_cylinders",
                "facility_facility"."phone_number",
                "facility_facility"."corona_testing",
                "facility_facility"."created_by_id",
                "facility_facility"."cover_image_url",
                "facility_facility"."middleware_address",
                "users_ward"."id",
                "users_ward"."local_body_id",
                "users_ward"."name",
                "users_ward"."number",
                "users_localbody"."id",
                "users_localbody"."district_id",
                "users_localbody"."name",
                "users_localbody"."body_type",
                "users_localbody"."localbody_code",
                "users_district"."id",
                "users_district"."state_id",
                "users_district"."name",
                "users_state"."id",
                "users_state"."name",
                T12."id",
                T12."local_body_id",
                T12."name",
                T12."number",
                T13."id",
                T13."district_id",
                T13."name",
                T13."body_type",
                T13."localbody_code",
                T14."id",
                T14."state_id",
                T14."name",
                T15."id",
                T15."name",
                T16."id",
                T16."password",
                T16."last_login",
                T16."is_superuser",
                T16."first_name",
                T16."last_name",
                T16."email",
                T16."is_staff",
                T16."is_active",
                T16."date_joined",
                T16."external_id",
                T16."username",
                T16."user_type",
                T16."created_by_id",
                T16."ward_id",
                T16."local_body_id",
                T16."district_id",
                T16."state_id",
                T16."phone_number",
                T16."alt_phone_number",
                T16."video_connect_link",
                T16."gender",
                T16."age",
                T16."home_facility_id",
                T16."weekly_working_hours",
                T16."doctor_qualification",
                T16."doctor_experience_commenced_on",
                T16."doctor_medical_council_registration",
                T16."verified",
                T16."deleted",
                T16."pf_endpoint",
                T16."pf_p256dh",
                T16."pf_auth",
                T16."asset_id",
                T17."id",
                T17."password",
                T17."last_login",
                T17."is_superuser",
                T17."first_name",
                T17."last_name",
                T17."email",
                T17."is_staff",
                T17."is_active",
                T17."date_joined",
                T17."external_id",
                T17."username",
                T17."user_type",
                T17."created_by_id",
                T17."ward_id",
                T17."local_body_id",
                T17."district_id",
                T17."state_id",
                T17."phone_number",
                T17."alt_phone_number",
                T17."video_connect_link",
                T17."gender",
                T17."age",
                T17."home_facility_id",
                T17."weekly_working_hours",
                T17."doctor_qualification",
                T17."doctor_experience_commenced_on",
                T17."doctor_medical_council_registration",
                T17."verified",
                T17."deleted",
                T17."pf_endpoint",
                T17."pf_p256dh",
                T17."pf_auth",
                T17."asset_id",
                T5."id",
                T5."external_id",
                T5."created_date",
                T5."modified_date",
                T5."deleted",
                T5."patient_id",
                T5."patient_no",
                T5."facility_id",
                T5."deprecated_diagnosis",
                T5."deprecated_icd11_provisional_diagnoses",
                T5."deprecated_icd11_diagnoses",
                T5."deprecated_icd11_principal_diagnosis",
                T5."symptoms",
                T5."other_symptoms",
                T5."symptoms_onset_date",
                T5."deprecated_covid_category",
                T5."category",
                T5."examination_details",
                T5."history_of_present_illness",
                T5."treatment_plan",
                T5."consultation_notes",
                T5."course_in_facility",
                T5."investigation",
                T5."prescriptions",
                T5."procedure",
                T5."suggestion",
                T5."route_to_facility",
                T5."review_interval",
                T5."referred_to_id",
                T5."referred_to_external",
                T5."transferred_from_location_id",
                T5."referred_from_facility_id",
                T5."referred_from_facility_external",
                T5."referred_by_external",
                T5."is_readmission",
                T5."admitted",
                T5."encounter_date",
                T5."icu_admission_date",
                T5."discharge_date",
                T5."discharge_reason",
                T5."new_discharge_reason",
                T5."discharge_notes",
                T5."discharge_prescription",
                T5."discharge_prn_prescription",
                T5."death_datetime",
                T5."death_confirmed_doctor",
                T5."bed_number",
                T5."is_kasp",
                T5."kasp_enabled_date",
                T5."is_telemedicine",
                T5."last_updated_by_telemedicine",
                T5."assigned_to_id",
                T5."medico_legal_case",
                T5."deprecated_verified_by",
                T5."treating_physician_id",
                T5."created_by_id",
                T5."last_edited_by_id",
                T5."last_daily_round_id",
                T5."current_bed_id",
                T5."height",
                T5."weight",
                T5."operation",
                T5."special_instruction",
                T5."intubation_history",
                T5."prn_prescription",
                T5."discharge_advice",
                "users_user"."id",
                "users_user"."password",
                "users_user"."last_login",
                "users_user"."is_superuser",
                "users_user"."first_name",
                "users_user"."last_name",
                "users_user"."email",
                "users_user"."is_staff",
                "users_user"."is_active",
                "users_user"."date_joined",
                "users_user"."external_id",
                "users_user"."username",
                "users_user"."user_type",
                "users_user"."created_by_id",
                "users_user"."ward_id",
                "users_user"."local_body_id",
                "users_user"."district_id",
                "users_user"."state_id",
                "users_user"."phone_number",
                "users_user"."alt_phone_number",
                "users_user"."video_connect_link",
                "users_user"."gender",
                "users_user"."age",
                "users_user"."home_facility_id",
                "users_user"."weekly_working_hours",
                "users_user"."doctor_qualification",
                "users_user"."doctor_experience_commenced_on",
                "users_user"."doctor_medical_council_registration",
                "users_user"."verified",
                "users_user"."deleted",
                "users_user"."pf_endpoint",
                "users_user"."pf_p256dh",
                "users_user"."pf_auth",
                "users_user"."asset_id",
                T7."id",
                T7."password",
                T7."last_login",
                T7."is_superuser",
                T7."first_name",
                T7."last_name",
                T7."email",
                T7."is_staff",
                T7."is_active",
                T7."date_joined",
                T7."external_id",
                T7."username",
                T7."user_type",
                T7."created_by_id",
                T7."ward_id",
                T7."local_body_id",
                T7."district_id",
                T7."state_id",
                T7."phone_number",
                T7."alt_phone_number",
                T7."video_connect_link",
                T7."gender",
                T7."age",
                T7."home_facility_id",
                T7."weekly_working_hours",
                T7."doctor_qualification",
                T7."doctor_experience_commenced_on",
                T7."doctor_medical_council_registration",
                T7."verified",
                T7."deleted",
                T7."pf_endpoint",
                T7."pf_p256dh",
                T7."pf_auth",
                T7."asset_id"
FROM "facility_patientregistration"
         LEFT OUTER JOIN "facility_facility"
                         ON ("facility_patientregistration"."facility_id" = "facility_facility"."id")
         LEFT OUTER JOIN "facility_patientconsultation"
                         ON ("facility_patientregistration"."id" = "facility_patientconsultation"."patient_id")
         LEFT OUTER JOIN "facility_patientconsultation" T5
                         ON ("facility_patientregistration"."last_consultation_id" = T5."id")
         LEFT OUTER JOIN "users_user" ON (T5."assigned_to_id" = "users_user"."id")
         LEFT OUTER JOIN "users_user" T7 ON ("facility_patientregistration"."assigned_to_id" = T7."id")
         LEFT OUTER JOIN "users_ward" ON ("facility_facility"."ward_id" = "users_ward"."id")
         LEFT OUTER JOIN "users_localbody" ON ("facility_facility"."local_body_id" = "users_localbody"."id")
         LEFT OUTER JOIN "users_district" ON ("facility_facility"."district_id" = "users_district"."id")
         LEFT OUTER JOIN "users_state" ON ("facility_facility"."state_id" = "users_state"."id")
         LEFT OUTER JOIN "users_ward" T12 ON ("facility_patientregistration"."ward_id" = T12."id")
         LEFT OUTER JOIN "users_localbody" T13 ON ("facility_patientregistration"."local_body_id" = T13."id")
         LEFT OUTER JOIN "users_district" T14 ON ("facility_patientregistration"."district_id" = T14."id")
         LEFT OUTER JOIN "users_state" T15 ON ("facility_patientregistration"."state_id" = T15."id")
         LEFT OUTER JOIN "users_user" T16 ON ("facility_patientregistration"."last_edited_id" = T16."id")
         LEFT OUTER JOIN "users_user" T17 ON ("facility_patientregistration"."created_by_id" = T17."id")
WHERE (NOT "facility_patientregistration"."deleted" AND
       ("facility_patientregistration"."facility_id" IN (1) OR "facility_patientconsultation"."facility_id" IN (1) OR
        T5."assigned_to_id" = 3 OR "facility_patientregistration"."assigned_to_id" = 3))
```

</details>

Reply:
Tried doing `queryset.distinct("id")` instead, and here's the query of that:

<details>
<summary>SQL Query (`queryset.distinct('id')`)</summary>

```sql
SELECT DISTINCT ON ("facility_patientregistration"."id") "facility_patientregistration"."id",
                                                         "facility_patientregistration"."external_id",
                                                         "facility_patientregistration"."created_date",
                                                         "facility_patientregistration"."modified_date",
                                                         "facility_patientregistration"."deleted",
                                                         "facility_patientregistration"."source",
                                                         "facility_patientregistration"."facility_id",
                                                         "facility_patientregistration"."nearest_facility_id",
                                                         "facility_patientregistration"."meta_info_id",
                                                         "facility_patientregistration"."name",
                                                         "facility_patientregistration"."age",
                                                         "facility_patientregistration"."gender",
                                                         "facility_patientregistration"."phone_number",
                                                         "facility_patientregistration"."emergency_phone_number",
                                                         "facility_patientregistration"."address",
                                                         "facility_patientregistration"."permanent_address",
                                                         "facility_patientregistration"."pincode",
                                                         "facility_patientregistration"."date_of_birth",
                                                         "facility_patientregistration"."year_of_birth",
                                                         "facility_patientregistration"."nationality",
                                                         "facility_patientregistration"."passport_no",
                                                         "facility_patientregistration"."is_medical_worker",
                                                         "facility_patientregistration"."blood_group",
                                                         "facility_patientregistration"."contact_with_confirmed_carrier",
                                                         "facility_patientregistration"."contact_with_suspected_carrier",
                                                         "facility_patientregistration"."estimated_contact_date",
                                                         "facility_patientregistration"."past_travel",
                                                         "facility_patientregistration"."countries_travelled_old",
                                                         "facility_patientregistration"."countries_travelled",
                                                         "facility_patientregistration"."date_of_return",
                                                         "facility_patientregistration"."allergies",
                                                         "facility_patientregistration"."present_health",
                                                         "facility_patientregistration"."ongoing_medication",
                                                         "facility_patientregistration"."has_SARI",
                                                         "facility_patientregistration"."is_antenatal",
                                                         "facility_patientregistration"."last_menstruation_start_date",
                                                         "facility_patientregistration"."date_of_delivery",
                                                         "facility_patientregistration"."ward_old",
                                                         "facility_patientregistration"."ward_id",
                                                         "facility_patientregistration"."local_body_id",
                                                         "facility_patientregistration"."district_id",
                                                         "facility_patientregistration"."state_id",
                                                         "facility_patientregistration"."is_migrant_worker",
                                                         "facility_patientregistration"."disease_status",
                                                         "facility_patientregistration"."number_of_aged_dependents",
                                                         "facility_patientregistration"."number_of_chronic_diseased_dependents",
                                                         "facility_patientregistration"."last_edited_id",
                                                         "facility_patientregistration"."action",
                                                         "facility_patientregistration"."review_time",
                                                         "facility_patientregistration"."created_by_id",
                                                         "facility_patientregistration"."is_active",
                                                         "facility_patientregistration"."date_of_receipt_of_information",
                                                         "facility_patientregistration"."test_id",
                                                         "facility_patientregistration"."date_of_test",
                                                         "facility_patientregistration"."srf_id",
                                                         "facility_patientregistration"."test_type",
                                                         "facility_patientregistration"."allow_transfer",
                                                         "facility_patientregistration"."last_consultation_id",
                                                         "facility_patientregistration"."will_donate_blood",
                                                         "facility_patientregistration"."fit_for_blood_donation",
                                                         "facility_patientregistration"."village",
                                                         "facility_patientregistration"."designation_of_health_care_worker",
                                                         "facility_patientregistration"."instituion_of_health_care_worker",
                                                         "facility_patientregistration"."transit_details",
                                                         "facility_patientregistration"."frontline_worker",
                                                         "facility_patientregistration"."date_of_result",
                                                         "facility_patientregistration"."number_of_primary_contacts",
                                                         "facility_patientregistration"."number_of_secondary_contacts",
                                                         "facility_patientregistration"."is_vaccinated",
                                                         "facility_patientregistration"."number_of_doses",
                                                         "facility_patientregistration"."vaccine_name",
                                                         "facility_patientregistration"."covin_id",
                                                         "facility_patientregistration"."last_vaccinated_date",
                                                         "facility_patientregistration"."cluster_name",
                                                         "facility_patientregistration"."is_declared_positive",
                                                         "facility_patientregistration"."date_declared_positive",
                                                         "facility_patientregistration"."assigned_to_id",
                                                         "facility_patientregistration"."abha_number_id",
                                                         "facility_facility"."id",
                                                         "facility_facility"."external_id",
                                                         "facility_facility"."created_date",
                                                         "facility_facility"."modified_date",
                                                         "facility_facility"."deleted",
                                                         "facility_facility"."name",
                                                         "facility_facility"."is_active",
                                                         "facility_facility"."verified",
                                                         "facility_facility"."facility_type",
                                                         "facility_facility"."kasp_empanelled",
                                                         "facility_facility"."features",
                                                         "facility_facility"."longitude",
                                                         "facility_facility"."latitude",
                                                         "facility_facility"."pincode",
                                                         "facility_facility"."address",
                                                         "facility_facility"."ward_id",
                                                         "facility_facility"."local_body_id",
                                                         "facility_facility"."district_id",
                                                         "facility_facility"."state_id",
                                                         "facility_facility"."oxygen_capacity",
                                                         "facility_facility"."type_b_cylinders",
                                                         "facility_facility"."type_c_cylinders",
                                                         "facility_facility"."type_d_cylinders",
                                                         "facility_facility"."expected_oxygen_requirement",
                                                         "facility_facility"."expected_type_b_cylinders",
                                                         "facility_facility"."expected_type_c_cylinders",
                                                         "facility_facility"."expected_type_d_cylinders",
                                                         "facility_facility"."phone_number",
                                                         "facility_facility"."corona_testing",
                                                         "facility_facility"."created_by_id",
                                                         "facility_facility"."cover_image_url",
                                                         "facility_facility"."middleware_address",
                                                         "users_ward"."id",
                                                         "users_ward"."local_body_id",
                                                         "users_ward"."name",
                                                         "users_ward"."number",
                                                         "users_localbody"."id",
                                                         "users_localbody"."district_id",
                                                         "users_localbody"."name",
                                                         "users_localbody"."body_type",
                                                         "users_localbody"."localbody_code",
                                                         "users_district"."id",
                                                         "users_district"."state_id",
                                                         "users_district"."name",
                                                         "users_state"."id",
                                                         "users_state"."name",
                                                         T12."id",
                                                         T12."local_body_id",
                                                         T12."name",
                                                         T12."number",
                                                         T13."id",
                                                         T13."district_id",
                                                         T13."name",
                                                         T13."body_type",
                                                         T13."localbody_code",
                                                         T14."id",
                                                         T14."state_id",
                                                         T14."name",
                                                         T15."id",
                                                         T15."name",
                                                         T16."id",
                                                         T16."password",
                                                         T16."last_login",
                                                         T16."is_superuser",
                                                         T16."first_name",
                                                         T16."last_name",
                                                         T16."email",
                                                         T16."is_staff",
                                                         T16."is_active",
                                                         T16."date_joined",
                                                         T16."external_id",
                                                         T16."username",
                                                         T16."user_type",
                                                         T16."created_by_id",
                                                         T16."ward_id",
                                                         T16."local_body_id",
                                                         T16."district_id",
                                                         T16."state_id",
                                                         T16."phone_number",
                                                         T16."alt_phone_number",
                                                         T16."video_connect_link",
                                                         T16."gender",
                                                         T16."age",
                                                         T16."home_facility_id",
                                                         T16."weekly_working_hours",
                                                         T16."doctor_qualification",
                                                         T16."doctor_experience_commenced_on",
                                                         T16."doctor_medical_council_registration",
                                                         T16."verified",
                                                         T16."deleted",
                                                         T16."pf_endpoint",
                                                         T16."pf_p256dh",
                                                         T16."pf_auth",
                                                         T16."asset_id",
                                                         T17."id",
                                                         T17."password",
                                                         T17."last_login",
                                                         T17."is_superuser",
                                                         T17."first_name",
                                                         T17."last_name",
                                                         T17."email",
                                                         T17."is_staff",
                                                         T17."is_active",
                                                         T17."date_joined",
                                                         T17."external_id",
                                                         T17."username",
                                                         T17."user_type",
                                                         T17."created_by_id",
                                                         T17."ward_id",
                                                         T17."local_body_id",
                                                         T17."district_id",
                                                         T17."state_id",
                                                         T17."phone_number",
                                                         T17."alt_phone_number",
                                                         T17."video_connect_link",
                                                         T17."gender",
                                                         T17."age",
                                                         T17."home_facility_id",
                                                         T17."weekly_working_hours",
                                                         T17."doctor_qualification",
                                                         T17."doctor_experience_commenced_on",
                                                         T17."doctor_medical_council_registration",
                                                         T17."verified",
                                                         T17."deleted",
                                                         T17."pf_endpoint",
                                                         T17."pf_p256dh",
                                                         T17."pf_auth",
                                                         T17."asset_id",
                                                         T5."id",
                                                         T5."external_id",
                                                         T5."created_date",
                                                         T5."modified_date",
                                                         T5."deleted",
                                                         T5."patient_id",
                                                         T5."patient_no",
                                                         T5."facility_id",
                                                         T5."deprecated_diagnosis",
                                                         T5."deprecated_icd11_provisional_diagnoses",
                                                         T5."deprecated_icd11_diagnoses",
                                                         T5."deprecated_icd11_principal_diagnosis",
                                                         T5."symptoms",
                                                         T5."other_symptoms",
                                                         T5."symptoms_onset_date",
                                                         T5."deprecated_covid_category",
                                                         T5."category",
                                                         T5."examination_details",
                                                         T5."history_of_present_illness",
                                                         T5."treatment_plan",
                                                         T5."consultation_notes",
                                                         T5."course_in_facility",
                                                         T5."investigation",
                                                         T5."prescriptions",
                                                         T5."procedure",
                                                         T5."suggestion",
                                                         T5."route_to_facility",
                                                         T5."review_interval",
                                                         T5."referred_to_id",
                                                         T5."referred_to_external",
                                                         T5."transferred_from_location_id",
                                                         T5."referred_from_facility_id",
                                                         T5."referred_from_facility_external",
                                                         T5."referred_by_external",
                                                         T5."is_readmission",
                                                         T5."admitted",
                                                         T5."encounter_date",
                                                         T5."icu_admission_date",
                                                         T5."discharge_date",
                                                         T5."discharge_reason",
                                                         T5."new_discharge_reason",
                                                         T5."discharge_notes",
                                                         T5."discharge_prescription",
                                                         T5."discharge_prn_prescription",
                                                         T5."death_datetime",
                                                         T5."death_confirmed_doctor",
                                                         T5."bed_number",
                                                         T5."is_kasp",
                                                         T5."kasp_enabled_date",
                                                         T5."is_telemedicine",
                                                         T5."last_updated_by_telemedicine",
                                                         T5."assigned_to_id",
                                                         T5."medico_legal_case",
                                                         T5."deprecated_verified_by",
                                                         T5."treating_physician_id",
                                                         T5."created_by_id",
                                                         T5."last_edited_by_id",
                                                         T5."last_daily_round_id",
                                                         T5."current_bed_id",
                                                         T5."height",
                                                         T5."weight",
                                                         T5."operation",
                                                         T5."special_instruction",
                                                         T5."intubation_history",
                                                         T5."prn_prescription",
                                                         T5."discharge_advice",
                                                         "users_user"."id",
                                                         "users_user"."password",
                                                         "users_user"."last_login",
                                                         "users_user"."is_superuser",
                                                         "users_user"."first_name",
                                                         "users_user"."last_name",
                                                         "users_user"."email",
                                                         "users_user"."is_staff",
                                                         "users_user"."is_active",
                                                         "users_user"."date_joined",
                                                         "users_user"."external_id",
                                                         "users_user"."username",
                                                         "users_user"."user_type",
                                                         "users_user"."created_by_id",
                                                         "users_user"."ward_id",
                                                         "users_user"."local_body_id",
                                                         "users_user"."district_id",
                                                         "users_user"."state_id",
                                                         "users_user"."phone_number",
                                                         "users_user"."alt_phone_number",
                                                         "users_user"."video_connect_link",
                                                         "users_user"."gender",
                                                         "users_user"."age",
                                                         "users_user"."home_facility_id",
                                                         "users_user"."weekly_working_hours",
                                                         "users_user"."doctor_qualification",
                                                         "users_user"."doctor_experience_commenced_on",
                                                         "users_user"."doctor_medical_council_registration",
                                                         "users_user"."verified",
                                                         "users_user"."deleted",
                                                         "users_user"."pf_endpoint",
                                                         "users_user"."pf_p256dh",
                                                         "users_user"."pf_auth",
                                                         "users_user"."asset_id",
                                                         T7."id",
                                                         T7."password",
                                                         T7."last_login",
                                                         T7."is_superuser",
                                                         T7."first_name",
                                                         T7."last_name",
                                                         T7."email",
                                                         T7."is_staff",
                                                         T7."is_active",
                                                         T7."date_joined",
                                                         T7."external_id",
                                                         T7."username",
                                                         T7."user_type",
                                                         T7."created_by_id",
                                                         T7."ward_id",
                                                         T7."local_body_id",
                                                         T7."district_id",
                                                         T7."state_id",
                                                         T7."phone_number",
                                                         T7."alt_phone_number",
                                                         T7."video_connect_link",
                                                         T7."gender",
                                                         T7."age",
                                                         T7."home_facility_id",
                                                         T7."weekly_working_hours",
                                                         T7."doctor_qualification",
                                                         T7."doctor_experience_commenced_on",
                                                         T7."doctor_medical_council_registration",
                                                         T7."verified",
                                                         T7."deleted",
                                                         T7."pf_endpoint",
                                                         T7."pf_p256dh",
                                                         T7."pf_auth",
                                                         T7."asset_id"
FROM "facility_patientregistration"
         LEFT OUTER JOIN "facility_facility"
                         ON ("facility_patientregistration"."facility_id" = "facility_facility"."id")
         LEFT OUTER JOIN "facility_patientconsultation"
                         ON ("facility_patientregistration"."id" = "facility_patientconsultation"."patient_id")
         LEFT OUTER JOIN "facility_patientconsultation" T5
                         ON ("facility_patientregistration"."last_consultation_id" = T5."id")
         LEFT OUTER JOIN "users_user" ON (T5."assigned_to_id" = "users_user"."id")
         LEFT OUTER JOIN "users_user" T7 ON ("facility_patientregistration"."assigned_to_id" = T7."id")
         LEFT OUTER JOIN "users_ward" ON ("facility_facility"."ward_id" = "users_ward"."id")
         LEFT OUTER JOIN "users_localbody" ON ("facility_facility"."local_body_id" = "users_localbody"."id")
         LEFT OUTER JOIN "users_district" ON ("facility_facility"."district_id" = "users_district"."id")
         LEFT OUTER JOIN "users_state" ON ("facility_facility"."state_id" = "users_state"."id")
         LEFT OUTER JOIN "users_ward" T12 ON ("facility_patientregistration"."ward_id" = T12."id")
         LEFT OUTER JOIN "users_localbody" T13 ON ("facility_patientregistration"."local_body_id" = T13."id")
         LEFT OUTER JOIN "users_district" T14 ON ("facility_patientregistration"."district_id" = T14."id")
         LEFT OUTER JOIN "users_state" T15 ON ("facility_patientregistration"."state_id" = T15."id")
         LEFT OUTER JOIN "users_user" T16 ON ("facility_patientregistration"."last_edited_id" = T16."id")
         LEFT OUTER JOIN "users_user" T17 ON ("facility_patientregistration"."created_by_id" = T17."id")
WHERE (NOT "facility_patientregistration"."deleted" AND
       ("facility_patientregistration"."facility_id" IN (1) OR "facility_patientconsultation"."facility_id" IN (1) OR
        T5."assigned_to_id" = 3 OR "facility_patientregistration"."assigned_to_id" = 3))
```

</details>